src/cdk/testing/tests/cross-environment.spec.ts

```diff
@@ -355,6 +366,68 @@ export function crossEnvironmentSpecs(
       expect(await getActiveElementId()).toBe(await input.getAttribute('id'));
     });
 
+    it('should be able to send key to a contenteditable', async () => {
+      const editable = await harness.editable();
+      const editableInputEvent = await harness.editableInputEvent();
+
+      await editable.sendKeys('Yi');
+
+      expect(await editable.text()).toBe('Yi');
+      expect(await editableInputEvent.text()).toBe('Count: 2');
+
+      await editable.clear();
+
+      expect(await editable.text()).toBe('');
+      expect(await editableInputEvent.text()).toBe('Count: 3');
+    });
+
+    it('should be able to send key to a child of a contenteditable', async () => {
+      const editable = await harness.editableP();
+
+      await editable.sendKeys('Yi');
+
+      expect(await editable.text()).toBe('Yi');
+    });
+
+    it('should not update not contenteditable div on send key', async () => {
+      const notEditable = await harness.notEditable();
+      const notEditableInputEvent = await harness.notEditableInputEvent();
+
+      await expectAsyncError(notEditable.sendKeys('Yi'));
+
+      expect(await notEditable.text()).toBe('');
+      expect(await notEditableInputEvent.text()).toBe('Count: 0');
+
+      await expectAsyncError(notEditable.clear());
+    });
+
+    it('should send key based on designMode', async () => {
+      const notEditable = await harness.notEditable();
+      const inheritEditable = await harness.inheritEditable();
+
+      await expectAsyncError(notEditable.sendKeys('Yi'));
+      expect(await notEditable.text()).toBe('');
+
+      await expectAsyncError(inheritEditable.sendKeys('Yi'));
+      expect(await inheritEditable.text()).toBe('');
+
+      await (await harness.designModeOnButton()).click();
+
+      await expectAsyncError(notEditable.sendKeys('Yi'));
+      expect(await notEditable.text()).toBe('');
+
+      await inheritEditable.sendKeys('Yi');
+      // The following expectation is failing on Protractor environment
+      // In protractor environment, the text is empty.
+      if (environment == TestEnvironment.TEST_BED) {
```

Comment:
We need to figure out why this is failing, not just disable the test

Reply:
Yeah, I understand that. Just wondering if you have any idea why this may be happening.
And even with this happening, if we can't find a solution, can we submit this CL?
(The CL is an improvement over current testbed implementation, this bug already exists in Protractor)

Reply:
I'd like to at least understand what's happening first. If we know why its happening I might be ok with submitting it as is (with a comment explaining).

Does the example work if you run it and manually interact? I'm not sure why its happening, but here are some guesses:
- Is the click handler on the design mode on button actually firing? Clicking is more finicky in Protractor, it can fail if other elements are overlapping or something like that
- Maybe it needs some delay before trying to edit? I'm not sure if the browser takes some time to activate design mode

Reply:
Sorry I am new here. How can I run the e2e test (as opposed to test) to see what's happening?

Reply:
You should be able to run the end to end tests with `yarn e2e` and the unit tests with `yarn test all`